tests/benchmarks/test_array.py

```diff
@@ -22,16 +22,15 @@ def print_size_info(memory: int, target_nbytes: int, *arrs: da.Array) -> None:
         )
 
 
-def test_anom_mean(small_client):
+def test_anom_mean(small_client, cluster_memory):
     # From https://github.com/dask/distributed/issues/2602#issuecomment-498718651
 
-    memory = cluster_memory(small_client)  # 76.66 GiB
-    target_nbytes = memory // 2
+    target_nbytes = cluster_memory // 2
     data = da.random.random(
         scaled_array_shape(target_nbytes, ("x", "10MiB")),
         chunks=(1, parse_bytes("10MiB") // 8),
     )
-    print_size_info(memory, target_nbytes, data)
+    print_size_info(cluster_memory, target_nbytes, data)
```

Comment:
nit: I don't love implementing this by lying to the test about the amount of cluster memory. It makes this logging of size info wrong, for one. It also means that tests that do want to adapt to cluster memory, but don't make sense to run on a variety of parameterizations (maybe they're meant to test spilling, or they know they won't work on larger datasets [see comment below]), are difficult to write.

Instead, why not have `memory_factor` be the fixture, and multiply within the test when appropriate? A little more code, but also more explicit and readable.

Reply:
+1